modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2109,20 +2196,26 @@ protected void addProperties(Map<String, Schema> properties, List<String> requir
         if (schema instanceof ComposedSchema) {
             ComposedSchema composedSchema = (ComposedSchema) schema;
 
-            for (Schema component : composedSchema.getAllOf()) {
-                addProperties(properties, required, component);
+            if (composedSchema.getAllOf() != null) {
```

Comment:
This is needed if a model omits allOf but includes anyOf or oneOf

Reply:
If it's oneOf/anyOf, `getAllOf` should return null or empty list, right? 

We didn't include the null check before and I don't recall issues related to this in particular but it could be an edge case we've not encountered yet.

Reply:
My new test cases include oneOf/anyOf with no allOf, and when that happens composedSchema.getAllOf() == null

Reply:
👌